pyrdp/security/nla.py

```diff
@@ -22,7 +24,7 @@ class NLAHandler(SegmentationObserver):
     This also logs the hash of NLA connection attempts.
     """
 
-    def __init__(self, sink: IntermediateLayer, state: NTLMSSPState, log: logging.LoggerAdapter):
+    def __init__(self, sink: IntermediateLayer, state: NTLMSSPState, log: logging.LoggerAdapter, ntlmCatch: bool = False):
```

Comment:
`ntlmCatch` here should probably be turned into a keyword argument. For example, if we want to add NTLM relay capability here, the current api would look like:

```
NLAHandler(sink, state, log, False, True)
```
The above being `ntlmCatch=False`, `ntlmRelay=True`.

Whereas if we use keyword arguments it could be:
```
NLAHandler(sink, state, log, relay=True)
```

Maybe using keyword arguments is too complex for a simple use case like that. I wonder what pattern already exists in PyRDP that could allow this Object to be extended cleanly. We need to check.

Also lets change `ntlmCatch` for `ntlmCapture` or `captureNtlm` idk.

Reply:
Please, check commit 20fa506 and let me know what you think.

Reply:
What I don't like about it is that it bundles PyRDP's own state into a class that tracks a pure protocol state. Also, since its impact is tested only on a single `if` and doesn't really need to be passed around, I feel like this is too drastic. That an argument on NLAHandler is enough.

Reply:
Reverted 20fa506 and using keyword to make it more clear that we are capturing the NTLM hash.